src/Uno.UI/UI/Xaml/Controls/Border/Border.cs

```diff
@@ -100,22 +100,40 @@ public UIElement Child
 				// disable the property-based propagation here to support the case where the Parent property is overridden to simulate
 				// a different inheritance hierarchy, as is done for some controls with native styles.
 				FrameworkPropertyMetadataOptions.ValueDoesNotInheritDataContext,
-				(s, e) => ((Border)s)?.OnChildChanged((UIElement)e.OldValue, (UIElement)e.NewValue)
+				(DependencyObject s, DependencyPropertyChangedEventArgs e) => ((Border)s)?.OnChildChanged((UIElement)e.OldValue, (UIElement)e.NewValue),
+				OnCoerceChild
```

Comment:
I personally would love if we can avoid coercion here and find a different way

Reply:
Fair, it sounds like we should be able to do this, as when the `Child` is set via XAML, it will be from `null` to the reference, so it should be fine